app/components/goodcity/code-search-overlay.js

```diff
@@ -0,0 +1,133 @@
+import Ember from "ember";
+import _ from "lodash";
+import SearchMixin from "stock/mixins/search_resource";
+
+/**
+ * An overlay that pops up from the bottom of the screen, allowing the user
+ * to search and select an order.
+ *
+ * The popup *does not* do anythinng to the order apart from returning it
+ *
+ * @property {boolean} open whether the popup is visible or not
+ * @property {function} onSelect callback triggered when an order is selected
+ */
+export default Ember.Component.extend(SearchMixin, {
+  fetchMoreResult: true,
+  store: Ember.inject.service(),
+  filter: "",
+  searchText: "",
+  fetchMoreResult: true,
+  storageType: null,
+  isSearchCodePreviousRoute: Ember.computed.localStorage(),
+
+  init() {
+    this._super("code-search-overlay");
+  },
+
+  allPackageTypes: Ember.computed("open", function() {
+    return this.get("store").peekAll("code");
+  }),
+
+  filteredPackageTypes: Ember.computed("allPackageTypes", function() {
+    const pkgTypes = this.get("allPackageTypes");
+    switch (this.get("storageType")) {
```

Comment:
this can be refactored as

```suggestion
    const pkgTypes = this.get("allPackageTypes");
    const storageType = this.get("storageType");
    return storageType ? pkgTypes.filterBy(`allow_${storageType.toLowerCase()}`, true) :  pkgTypes;
```